msal/tests/Test.MSAL.NET.Unit.net45/UnifiedCacheTests.cs

```diff
@@ -131,6 +124,92 @@ public void UnifiedCache_MsalStoresToAndReadRtFromAdalCache()
+                };
+
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                MsalMockHelpers.ConfigureMockWebUI(new AuthorizationResult(AuthorizationStatus.Success,
+                                                                           app.RedirectUri + "?code=some-code"));
+
+                httpManager.AddMockHandlerForTenantEndpointDiscovery(MsalTestConstants.AuthorityHomeTenant);
+                httpManager.AddSuccessTokenResponseMockHandlerForPost();
+
+                AuthenticationResult result = app.AcquireTokenAsync(MsalTestConstants.Scope).Result;
+                Assert.IsNotNull(result);
+
+                Assert.AreEqual(1, tokenCache.tokenCacheAccessor.GetAllAccountsAsString().Count);
+                Assert.AreEqual(1, app.GetAccountsAsync().Result.Count());
+
+                // login tp app1 with same credentials
+                var tokenCache1 = new TokenCache();
+                tokenCache1.SetBeforeAccess((TokenCacheNotificationArgs args) =>
+                {
+                    args.TokenCache.Deserialize(data);
+                });
+                tokenCache1.SetAfterAccess((TokenCacheNotificationArgs args) =>
+                {
+                    data = args.TokenCache.Serialize();
+                });
+
+                PublicClientApplication app1 = new PublicClientApplication(
+                    httpManager,
+                    MsalTestConstants.ClientId_1,
+                    ClientApplicationBase.DefaultAuthority)
+                {
+                    UserTokenCache = tokenCache1
+                };
+
+                MsalMockHelpers.ConfigureMockWebUI(new AuthorizationResult(AuthorizationStatus.Success,
+                                                                           app.RedirectUri + "?code=some-code"));
+                httpManager.AddSuccessTokenResponseMockHandlerForPost();
+
+                result = app1.AcquireTokenAsync(MsalTestConstants.Scope).Result;
+                Assert.IsNotNull(result);
+
+                // make sure that only one account cache entity was created
+                Assert.AreEqual(1, tokenCache1.tokenCacheAccessor.GetAllAccountsAsString().Count);
+                Assert.AreEqual(1, app1.GetAccountsAsync().Result.Count());
+
+                Assert.AreEqual(2, tokenCache1.tokenCacheAccessor.GetAllAccessTokensAsString().Count);
+                Assert.AreEqual(2, tokenCache1.tokenCacheAccessor.GetAllRefreshTokensAsString().Count);
+                Assert.AreEqual(2, tokenCache1.tokenCacheAccessor.GetAllIdTokensAsString().Count);
+
+                // remove account from app
+                app.RemoveAsync(app.GetAccountsAsync().Result.First()).Wait();
+
+                // make sure account removed from app
+                Assert.AreEqual(0, app.GetAccountsAsync().Result.Count());
+
+                // make sure account Not removed from app1
+                Assert.AreEqual(1, app1.GetAccountsAsync().Result.Count());
```

Comment:
do you want to make sure you can remove the token from app1 too?
`Assert.AreEqual(0, app1.GetAccountsAsync().Result.Count()); `

Reply:
yes, i check remove account for app return 0 account but for app1 1 account - so remove call on app did not affect app1